dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/consumer/TaskPriorityQueueConsumer.java

```diff
@@ -151,6 +173,10 @@ public List<TaskPriority> batchDispatch(int fetchTaskNum) throws TaskPriorityQue
         List<TaskPriority> failedDispatchTasks = Collections.synchronizedList(new ArrayList<>());
         CountDownLatch latch = new CountDownLatch(fetchTaskNum);
 
+        if (taskPriorityDispatchFailedQueue.size() > 0) {
+            dispatchFailedBackToTaskPriorityQueue(fetchTaskNum);
```

Comment:
Will it take affect to the performance if handle in the same thread?

Reply:
It would be better to have a separate thread for the failed queue retry.

Reply:
The default task scheduling thread pool size is 2, which ensures that the processing in the failure queue and the scheduling queue do not affect each other.
In other cases add the default scheduling thread pool size +1.